rtloader/common/builtins/datadog_agent.c

```diff
@@ -653,3 +662,46 @@ static PyObject *set_external_tags(PyObject *self, PyObject *args)
     Py_RETURN_NONE;
 
 }
+
+/*! \fn PyObject *obfuscate_sql(PyObject *self, PyObject *args)
+    \brief This function implements the `datadog_agent.obfuscate_sql` method, obfuscating
+    the provided sql string.
+    \param self A PyObject* pointer to the `datadog_agent` module.
+    \param args A PyObject* pointer to a tuple containing the key to retrieve.
+    \return A PyObject* pointer to the value.
+
+    This function is callable as the `datadog_agent.obfuscate_sql` Python method and
+    uses the `cb_obfuscate_sql()` callback to retrieve the value from the agent
+    with CGO. If the callback has not been set `None` will be returned.
+*/
+static PyObject *obfuscate_sql(PyObject *self, PyObject *args)
+{
+    // callback must be set
+    if (cb_obfuscate_sql == NULL) {
+        Py_RETURN_NONE;
+    }
+
+    PyGILState_STATE gstate = PyGILState_Ensure();
+
+    char *rawQuery;
+    if (!PyArg_ParseTuple(args, "s", &rawQuery)) {
+        return NULL;
+    }
+
+    char *obfQuery = NULL;
+    char *err = NULL;
+    obfQuery = cb_obfuscate_sql(rawQuery, &err);
+
+    PyObject *retval = NULL;
+    if (err != NULL) {
+        PyErr_SetString(PyExc_RuntimeError, err);
+        cgo_free(err);
+    }
```

Comment:
If both `err` and `obfQuery` are not null we will set an error that will be raise the next time a Cpython function return null.
I think if `err` is not null we should raise the exception (free `obfQuery` of needed) and return NULL.

Reply:
Yes good point! Updated to return null if error, regardless of obfQuery result. Does that work?